### PR TITLE
EL-358 - ListJoinedDate support

### PR DESCRIPTION
**Pull request by ijer**
This adds support to for the two new subscriber properties in v3.3: ListJoinedDate and StateChangedDate

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -14,6 +14,14 @@
   * Campaigns can be filtered by Sent Date using sentFromDate and sentToDate in YYYY-MM-DD format
   * Campaigns can be filtered by their tags.
 * Campaign summary() now returns Name
+* Following endpoints now return ListJoinedDate for each subscriber.  Also, Date has been renamed to StateChangedDate:
```
Comment (henrys-cm) [inline HISTORY.md:17]: Potentially marking this as 'breaking' changes (the rename of the date field)

**Review by henrys-cm — APPROVED** [review]
LGTM -- (optional -- adding the other samples for consistency)

File: src/com/createsend/models/subscribers/SubscriberWithJoinedDate.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package com.createsend.models.subscribers;
+
+import java.util.Date;
+
+public class SubscriberWithJoinedDate {
```
Comment (henrys-cm) [inline src/com/createsend/models/subscribers/SubscriberWithJoinedDate.java:26]: Just a thought if we just update the Subscriber object altogether (ie: the wrapper doesn't need to do backward compatibility)?

File: samples/com/createsend/samples/SampleRunner.java [hunk 3]
```diff
@@ -314,6 +321,30 @@ private static void runListMethods(String clientID) throws CreateSendException {
...
+        Date subscribersFrom = new Date();
+
+        System.out.printf("Result of list active: %s\n",
+                listAPI.active(subscribersFrom, null, null, null, null));
+        System.out.printf("Result of list unsubscribed: %s\n",
+                listAPI.unsubscribed(subscribersFrom, null, null, null, null));
+        System.out.printf("Result of list bounced: %s\n",
+                listAPI.bounced(subscribersFrom, null, null, null, null));
+        System.out.printf("Result of list deleted: %s\n",
+                listAPI.deleted(subscribersFrom, null, null, null, null));
```
[the remark this answers is not in the document]
Comment (ijer) [reply samples/com/createsend/samples/SampleRunner.java:336]: yes, good point Henry, I'll add unconfirmed.  subscribers-details is already in another sample so I won't add it again.

File: src/com/createsend/models/subscribers/SubscriberWithJoinedDate.java (hunk 2 above)
Comment (ijer) [reply src/com/createsend/models/subscribers/SubscriberWithJoinedDate.java:26]: Although most usages of Subscriber Date has been renamed to `StateChangedDate`, we're still using the Subscriber entity with Date in some places (e.g. Campaign unsubscribes)

**Comment by ijer** [comment]
Thanks for the review @henrys-cm .  It's ready for re-review.  Let me know what you think of this.

**Comment by ijer** [comment]
Thanks for the updates @henrys-cm

**Pull request merged**